### PR TITLE
support navigation to hash url

### PR DESCRIPTION
**Pull request by nadav-ab**
This PR changes the navigation to include the `search` and the `hash` alongside `pathname` in the navigation uri.

### CODE REVIEW

File: packages/define-remix-app/src/define-remix-app.tsx [hunk 1]
```diff
@@ -205,17 +207,26 @@ export default function defineRemixApp({ appPath, routingPattern = 'file' }: IDe
         }: IReactAppProps<RouteExtraInfo>) => {
             const uriRef = useRef(uri);
             uriRef.current = uri;
-            const { Router, navigate } = useMemo(
-                () => manifestToRouter(manifest, importModule, setUri, onCaughtError, uriRef, callServerMethod),
+            const { Router } = useMemo(
```
Comment (PeterShershov) [inline packages/define-remix-app/src/define-remix-app.tsx:210]: do we still need to return the `navigate` method from manifestToRouter?

File: packages/define-remix-app/src/manifest-to-router.tsx [hunk 2]
```diff
@@ -143,29 +149,43 @@ export const fileToRoute = (
...
 }) {
     const currentModule = useDispatcher(module);
-
-    const uri = useLocation().pathname;
-
     navigation.setNavigateFunction(useNavigate());
 
+    const location = useLocation();
+
+    const uri = locationToUri(location);
```
Comment (PeterShershov) [inline packages/define-remix-app/src/manifest-to-router.tsx:188]:
```ts
const { pathname, search = '', hash = '' } = useLocation();
const uri = `${pathname}${search}${hash}`;
```

**Review by PeterShershov — COMMENTED** [review]
can we test this?

**Review by PeterShershov — APPROVED** [review]

**Comment by idoros** [comment]
> can we test this?

I assume you could, why isn't there a test?

**Pull request merged**